### PR TITLE
Changes to support the new RunConfig pattern in core

### PR DESCRIPTION
**Pull request by jcrist**
Adds a new `flow.run_config` field that can be used instead of `flow.environment` to configure a flow run. There's likely more to do here, but so far this works with https://github.com/PrefectHQ/prefect/pull/3333, see that PR for the full motivation.

### CODE REVIEW

File: src/prefect_server/api/flows.py [hunk 1]
```diff
@@ -208,7 +210,8 @@ async def create_flow(
         name=flow.name,
         serialized_flow=serialized_flow,
         environment=flow.environment,
-        core_version=flow.environment.get("__version__"),
+        run_config=flow.run_config,
+        core_version=flow.__version__,
```
Comment (cicdw) [inline src/prefect_server/api/flows.py:214]: I think we need `flow.__version__ or flow.environment.get("__version__")` for backwards compatibility

**Review by cicdw — COMMENTED** [review]
I realize this won't be fully ready until we merge the Core PR, but this looks straightforward to me!  I left one comment on backwards compatibility

File: src/prefect_server/api/flows.py (hunk 1 above)
Comment (jcrist) [reply src/prefect_server/api/flows.py:214]: All flows have a `__version__` attached already (and have for a while), it's part of the base `ObjectSchema` serializer. I don't think a backwards compat fix is needed here, but might be wrong.

**Comment by jcrist** [comment]
I'm seeing errors about `null` run-config values, should that field be nullable (currently it's not, but afaict the server-side default should be being applied)?

File: src/prefect_server/api/flows.py (hunk 1 above)
Comment (cicdw) [reply src/prefect_server/api/flows.py:214]: Ahh you're right

**Comment by cicdw** [comment]
@jcrist the `server_default` is only used if you don't provide a value for `run_config` when inserting a new row.  But our ORM will always provide a value (and in this case it's providing `null`).  If you want to keep the non-nullability, you'll need to put a `run_config or {}` in the model before the insert.

**Comment by jcrist** [comment]
I believe this is good to go.

**Review by cicdw — DISMISSED** [review]
LGTM!  Welcome to the server repo @jcrist :D

**Review by cicdw — APPROVED** [review]

**Pull request merged**